### PR TITLE
feat: workshop details page + manfred steyer 2024 workshop #75

### PR DESCRIPTION
**Pull request by peppoasap**
#### Changes Made:

1. **Workshop Detail Page:**
   - Created a new workshop detail page for enhanced user experience.
   - Implemented a detailed structure for workshop details, including title, description, date, author information, location details, image, link, and additional custom attributes.

2. **Workshop Model Updates:**
   - Modified the workshop model to include a more detailed structure, enhancing the workshop attributes for better representation and utilization.

   ```typescript
   export interface WorkshopAttributes {
     slug: string;
     tag: string;
     title: string;
     description: string;
     date: string;
     author: {
       name: string;
       biography: string;
       image: string;
       link: string;
     };
     location: {
       name: string;
       mapsLink: string;
     };
     image: string;
     link: string;
     col: number;
     [key: string]: string | number | boolean | Date | undefined | Object;
   }
   ```

3. **Sorting and Filtering:**
   - Implemented sorting and filtering(just internally, not for user) of past workshops based on the workshop date.
   - Ensured that only workshops with dates prior to the current date are considered as past workshops.

4. **Query Params for Workshop Detail:**
   - Added query parameters to retrieve the correct workshop details by slug using the internal API.
   - Improved the retrieval process for workshop details on the workshop detail page.

5. **Additional Workshop:**
   - Added a new workshop for Manfred Steyer on 26 June.

#### How to Test:

1. Navigate to the workshops page.
2. Check the sorting and filtering of past workshops based on the workshop date.
3. Test the query parameter functionality by accessing specific workshop details clicking/using the slug.
4. Verify that the workshop details are displayed correctly with the new structure.

#### Additional Notes:

I need a bit of help to connect the tito ticket functionality and a review of the layout to understand if some changes are needed to be more inline with the current design.

### CODE REVIEW

**Comment by lucianomurr** [comment]
I get some layout errors for the detail page, is the same for you?

https://deploy-preview-76--ngrome.netlify.app/workshops/angular-architects-signal-ddd-mfe

**Comment by peppoasap** [comment]
> I get some layout errors for the detail page, is the same for you?
> 
> https://deploy-preview-76--ngrome.netlify.app/workshops/angular-architects-signal-ddd-mfe

Mmmh yes, seems is not loading the right workshop using the api filtered by slug, seems like the workshop api file is not updated...maybe there is something to specify due to SSR?

**Comment by lucianomurr** [comment]
Try to add the api and the workshop page inside the pre-rendered routes in `vite.config.ts`

https://github.com/ngromeconf/ngrome-website/blob/main/vite.config.ts

**Comment by peppoasap** [comment]
> Try to add the api and the workshop page inside the pre-rendered routes in `vite.config.ts`
> 
> https://github.com/ngromeconf/ngrome-website/blob/main/vite.config.ts

Tried to do some changes on config but didn't work, I'm looking for some solutions on docs, will try something different tonight

File: vite.config.ts [hunk 1]
```diff
@@ -23,6 +23,7 @@ export default defineConfig(({ mode }) => ({
           '/code-of-conduct',
           '/terms-and-privacy',
           '/workshops',
+          '/workshops/angular-architects-signal-ddd-mfe',
           '/api/v1/workshops',
```
Comment (lucianomurr) [inline vite.config.ts:27]: try to add also this one:
          `'/api/v1/workshops/angular-architects-signal-ddd-mfe',`

File: src/server/routes/v1/workshops/[slug].get.ts [hunk 2]
```diff
@@ -0,0 +1,8 @@
+import { defineEventHandler, getRouterParam } from 'h3';
+import { WORKSHOP } from '../workshops';
+
+export default defineEventHandler((event) => {
+  const slug = getRouterParam(event, 'slug');
+  if (!slug) return null;
```
Comment (lucianomurr) [inline src/server/routes/v1/workshops/[slug].get.ts:6]: A possible alternative could be to add some filter to the same workshop API, or filter directly in the `[workshops/[slug].page.ts](https://github.com/ngromeconf/ngrome-website/pull/76/files#diff-bfafb1a2c6105d61b7d92829ec5b1379c9fc7ad8341a5fe7c62adb7834fe806b)` (which could be not the optimal solution ) but should work!

**Comment by lucianomurr** [comment]
issue #75

**Comment by lucianomurr** [comment]
closes #75

**Pull request merged**